### PR TITLE
Feature - login prompt

### PR DESCRIPTION
**Pull request by xserxses**

### CODE REVIEW

File: webflows/build.gradle [hunk 1]
```diff
@@ -60,15 +65,18 @@ dependencies {
 
     api "androidx.browser:browser:1.4.0"
     implementation 'com.nimbusds:nimbus-jose-jwt:9.24.3'
+    implementation 'androidx.lifecycle:lifecycle-runtime-ktx:2.4.0'
 
     //Logging
     implementation "com.jakewharton.timber:timber:$timber_version"
+    implementation 'androidx.constraintlayout:constraintlayout:2.2.0-alpha09'
```
Comment (xserxses) [inline webflows/build.gradle:72]: [Alpha12](https://developer.android.com/jetpack/androidx/releases/constraintlayout#constraintlayout-2.2.0-alpha12) is now available

File: webflows/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -4,6 +4,15 @@
     <uses-permission android:name="android.permission.INTERNET" />
 
     <application>
+        <provider
+          android:name="com.schibsted.account.webflows.loginPrompt.LoginPromptContentProvider"
+          android:authorities="${applicationId}.contentprovider"
```
Comment (xserxses) [inline webflows/src/main/AndroidManifest.xml:9]: `${applicationId}` looks suspicious. Maybe thats' why Your testing with example app was not working?

https://developer.android.com/guide/topics/manifest/provider-element#auth

File: webflows/src/main/java/com/schibsted/account/webflows/activities/AuthorizationManagementActivity.kt [hunk 3]
```diff
@@ -177,6 +179,7 @@ class AuthorizationManagementActivity : Activity() {
 
     private fun handleAuthorizationCanceled() {
         Timber.d("Authorization flow canceled by user")
+        SchibstedAccountTracker.track(SchibstedAccountTrackingEvent.UserLoginCanceled)
```
Comment (xserxses) [inline webflows/src/main/java/com/schibsted/account/webflows/activities/AuthorizationManagementActivity.kt:182]: Note: This not be triggered if user cancel Eid Verification process by "Abryt" button

File: webflows/src/main/java/com/schibsted/account/webflows/loginPrompt/LoginPromptManager.kt [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    }
+
+    /**
+     * Show login prompt.
+     *
+     * @param supportFragmentManager Calling entity's fragment manager.
+     */
+    fun showLoginPrompt(supportFragmentManager: FragmentManager) {
+        var loginPromptFragment =
+            supportFragmentManager.findFragmentByTag(fragmentTag) as? LoginPromptFragment
```
Comment (xserxses) [inline webflows/src/main/java/com/schibsted/account/webflows/loginPrompt/LoginPromptManager.kt:31]: If You find by tag You still need to assign this field 

```
loginPromptFragment.loginPromptConfig = this.loginPromptConfig
```
As we discussed this fragment might be recreated

File: webflows/src/main/java/com/schibsted/account/webflows/loginPrompt/LoginPromptManager.kt [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+        this.client = client
+        this.isCancelable = isCancelable
+    }
+}
+
+internal class LoginPromptManager {
+    val loginPromptConfig: LoginPromptConfig
+    val fragmentTag = "schibsted_account_login_prompt"
+
+    constructor(loginPromptConfig: LoginPromptConfig) {
```
Comment (xserxses) [inline webflows/src/main/java/com/schibsted/account/webflows/loginPrompt/LoginPromptManager.kt:20]: You can have this constructor simplified
```
internal class LoginPromptManager(val loginPromptConfig: LoginPromptConfig)
```

fragmentTag - can be private const or assignable in constructor with default value.

File: webflows/src/main/java/com/schibsted/account/webflows/loginPrompt/LoginPromptManager.kt [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+    /**
+     * Show login prompt.
+     *
+     * @param supportFragmentManager Calling entity's fragment manager.
+     */
+    fun showLoginPrompt(supportFragmentManager: FragmentManager) {
+        var loginPromptFragment =
+            supportFragmentManager.findFragmentByTag(fragmentTag) as? LoginPromptFragment
+                ?: initializeLoginPrompt()
+        loginPromptFragment.show(supportFragmentManager, fragmentTag)
```
Comment (xserxses) [inline webflows/src/main/java/com/schibsted/account/webflows/loginPrompt/LoginPromptManager.kt:33]: Proabbly we'll need ` loginPromptFragment?.dialog?.isShowing ` check here before calling show.

File: webflows/src/main/java/com/schibsted/account/webflows/loginPrompt/SessionInfoManager.kt [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+import android.content.ContentValues
+import android.content.Context
+import android.content.Intent
+import android.content.pm.PackageManager
+import android.content.pm.ResolveInfo
+import android.net.Uri
+import android.os.Build
+
+
+internal class SessionInfoManager(context: Context) {
```
Comment (xserxses) [inline webflows/src/main/java/com/schibsted/account/webflows/loginPrompt/SessionInfoManager.kt:12]: We need to take a closer look on it..
Something is fishy or we are not sure how to use ContentProviders lol

We need to ceck it in hermes and look on linked manifest after building the app

File: webflows/src/main/java/com/schibsted/account/webflows/tracking/SchibstedAccountTrackingStore.kt [hunk 8]
```diff
@@ -0,0 +1,37 @@
+package com.schibsted.account.webflows.tracking
+
+interface SchibstedAccountTrackerStore {
+    fun addTrackingListener(trackingListener: SchibstedAccountTrackingListener)
+    fun removeTrackingListener(trackingListener: SchibstedAccountTrackingListener)
+
+    companion object : SchibstedAccountTrackerStore {
+        private val accountListenersList: MutableList<SchibstedAccountTrackingListener> =
```
Comment (xserxses) [inline webflows/src/main/java/com/schibsted/account/webflows/tracking/SchibstedAccountTrackingStore.kt:8]: Thread safety?
To consider

File: webflows/src/main/java/com/schibsted/account/webflows/util/Util.kt [hunk 9]
```diff
@@ -42,4 +45,11 @@ internal object Util {
 
         return "${split[0]}.${split[1]}"
     }
+
+    fun isCustomTabsSupported(context: Context): Boolean {
+        val serviceIntent = Intent(CustomTabsService.ACTION_CUSTOM_TABS_CONNECTION)
+        val resolveInfos = context.packageManager.queryIntentServices(serviceIntent, 0)
+
+        return !resolveInfos.isEmpty()
```
Comment (xserxses) [inline webflows/src/main/java/com/schibsted/account/webflows/util/Util.kt:53]:
```suggestion
        return resolveInfos.isNotEmpty()
```

**Review by xserxses — COMMENTED** [review]
Can't Request changes on my own PR ;)

But me and Rafał don't think it's prod ready now.

Although we are close :)

**Comment by xserxses** [comment]
@bogdan-niculescu-sch FYI

File: webflows/src/main/res/layout/login_prompt.xml [hunk 10]
```diff
@@ -0,0 +1,118 @@
+<?xml version="1.0" encoding="utf-8"?>
+<androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
```
[the remark this answers is not in the document]
Comment (bogdan-niculescu-sch) [reply webflows/src/main/res/layout/login_prompt.xml:2]: Think there is a small difference between the two layouts. Attaching comparison below
Screenshot 2023-09-04 at 11 18 52

I have also looked into simplifying the layout overall in the most recent changes

**Review by bogdan-niculescu-sch — APPROVED** [review]

**Pull request merged**